src/gltf-component-mappings.js

```diff
@@ -13,12 +13,30 @@ AFRAME.GLTFModelPlus.registerComponent("gltf-model-plus", "gltf-model-plus");
 AFRAME.GLTFModelPlus.registerComponent("media-loader", "media-loader");
 AFRAME.GLTFModelPlus.registerComponent("body", "body");
 AFRAME.GLTFModelPlus.registerComponent("hide-when-quality", "hide-when-quality");
-AFRAME.GLTFModelPlus.registerComponent("light", "light");
+AFRAME.GLTFModelPlus.registerComponent("light", "light", (el, componentName, componentData) => {
+  if (componentData.distance === 0) {
+    componentData.decay = 0;
+  }
+  el.setAttribute(componentName, componentData);
+  if (componentData.castShadow) {
+    // HACK: component.light isn't set until one frame after setArrtibute
+    setTimeout(() => {
+      el.components.light.light.shadow.camera.matrixNeedsUpdate = true;
+    }, 0);
+  }
+});
 AFRAME.GLTFModelPlus.registerComponent("ambient-light", "ambient-light");
 AFRAME.GLTFModelPlus.registerComponent("directional-light", "directional-light");
 AFRAME.GLTFModelPlus.registerComponent("hemisphere-light", "hemisphere-light");
-AFRAME.GLTFModelPlus.registerComponent("point-light", "point-light");
-AFRAME.GLTFModelPlus.registerComponent("spot-light", "spot-light");
+function decayMigration(el, componentName, componentData) {
+  if (componentData.range === 0) {
```

Comment:
Seems like an odd migration, Is it no longer valid to have a range of 0 and a decay of non 0?

Reply:
This reverts the old behavior of "range/distance" being 0 in ThreeJS. See https://github.com/mrdoob/three.js/pull/14962/files#diff-a63071d948a349b79bddc096ce9f35f4L215